shell.nix

```diff
@@ -0,0 +1,64 @@
+{
+  pkgs ? import <nixpkgs> { },
+}:
+
+pkgs.mkShell {
+  nativeBuildInputs =
+    with pkgs;
+    [
```

Comment:
Add `clang` and `mold` here. These packages are required because of https://github.com/zed-industries/zed/blob/main/.cargo/config.toml